hcl/hcl_test.go

```diff
@@ -50,6 +50,7 @@ func TestHCLParserModules(t *testing.T) {
 	type want struct {
 		modules []hcl.Module
 		err     error
```

Comment:
I think this should be removed and always treat as returning multiple errors. No?

Reply:
yeah but I didn't wanted to change all the tests also on this PR so we could focus more on the actual changes, will do in a follow up PR.